src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -174,6 +181,117 @@ public void connect() {
     this.connect(this.username, this.password, this.hostname, this.port);
   }
 
+  /**
+   * This function imports an image file to an omero server.
+   * The omero dataset to which the image is registered to is fetched based on the openBIS identifiers.
+   *
+   * @param filePath the path to the image file
+   * @param projectId the project ID in OpenBIS
+   * @param sampleCode the sample Code in OpenBIS
+   * @return newly generated omero IDs for registered images
+   */
+  public Set<Long> registerImageFile(String filePath, String projectId, String sampleCode) {
+    Long datasetId = findOmeroDataset(projectId, sampleCode);
+    if (datasetId == null) {
+      return new HashSet<>();
```

Comment:
wouldn't an error be more useful here?

Reply:
I think that it should be null if nothing was found. This can happen also with correct search criteria. We do not know whether a dataset is present or not.

Reply:
it's an empty set, though, meaning the return value has to be closely monitored when using this function. if this is done, that's okay, of course - it should also be mentioned in the header then, that the set can be empty.

Reply:
Ture. You are right, it should be consistent.